app/models/asset.rb

```diff
@@ -152,11 +179,12 @@ def self.search(
     end
 
     Asset
-      .joins("LEFT JOIN asset_text_data ON assets.id = asset_text_data.asset_id")
-      .select("assets.*")
-      .select("ts_headline(data, to_tsquery('" + s_query + "'),
-        'StartSel=<mark>, StopSel=</mark>') headline")
-      .where("assets.id IN (?)",  ids)
+      .joins('LEFT JOIN asset_text_data ON ' \
+             ' assets.id = asset_text_data.asset_id')
+      .select('assets.*')
+      .select("ts_headline(data, to_tsquery('" + s_query +
```

Comment:
Can you use ActiveRecord::Sanitization for s_query variable here or sanitize it before you do the trim/map/join stuff above?